### DISCUSSION TITLE
Offline Static Map

### DISCUSSION THREAD

**Discussion by JamesRexMiller4** [visgl/deck.gl#6987] [Q&A]
I am using Mapbox-GL-JS underneath deck.gl and wondering if it is possible to use this offline, either by self-hosting the vector tiles on localhost or providing a configured style JSON. 

Mapbox documentation is not totally clear; they have options for Android and iOS SDKSs but seemingly nothing for Mapbox-GL-JS.

**Comment by kylebarron** [comment]
This question doesn't seem related to deck.gl?

Anyways, Mapbox GL JS offline should work. You just need to self host the fonts, sprites, style json, and vector tile data. You should be able to find some answers online, e.g. https://stackoverflow.com/q/28330893. I have an old example [here](https://github.com/nst-guide/osm-liberty-topo/blob/gh-pages/style.json) of a style that doesn't touch Mapbox servers

**Reply by JamesRexMiller4** [reply]
Sorry, technically is more related to Mapbox than deck.gl, I just find the maintainers of this library to be far more responsive and helpful than Mapbox so figured I'd ask. 

And surprisingly there is not as clear of answers online as one would expect, so thanks for sharing your old example!

**Reply by kylebarron** [reply]
I think if you put all the resources on localhost it should work.
